src/main/java/org/opensearch/neuralsearch/processor/AgenticQueryTranslatorProcessor.java

```diff
@@ -0,0 +1,188 @@
+        if (!(query instanceof AgenticSearchQueryBuilder)) {
+            requestListener.onResponse(request);
+            return;
+        }
+
+        // Validate that agentic query is used alone without other search features
+        if (hasOtherSearchFeatures(sourceBuilder)) {
+            requestListener.onFailure(
+                new IllegalArgumentException(
+                    "Agentic search cannot be used with other search features like aggregations, sort, highlighters, etc."
+                )
+            );
+            return;
+        }
+
+        AgenticSearchQueryBuilder agenticQuery = (AgenticSearchQueryBuilder) query;
+        executeAgentAsync(agenticQuery, request, requestListener);
+    }
+
+    private boolean hasOtherSearchFeatures(SearchSourceBuilder sourceBuilder) {
+        return sourceBuilder.aggregations() != null
+            || sourceBuilder.sorts() != null && !sourceBuilder.sorts().isEmpty()
+            || sourceBuilder.highlighter() != null
+            || sourceBuilder.postFilter() != null
+            || sourceBuilder.suggest() != null
+            || sourceBuilder.rescores() != null && !sourceBuilder.rescores().isEmpty()
+            || sourceBuilder.collapse() != null;
+    }
+
+    private void executeAgentAsync(
+        AgenticSearchQueryBuilder agenticQuery,
+        SearchRequest request,
+        ActionListener<SearchRequest> requestListener
+    ) {
+        Map<String, String> parameters = new HashMap<>();
+        parameters.put("query_text", agenticQuery.getQueryText());
+
+        // Get index mapping from the search request
+        if (request.indices() != null && request.indices().length > 0) {
+            try {
+                String cacheKey = String.join(",", request.indices());
+                String cachedMappingJson = indexMappingCache.get(cacheKey);
+
+                if (cachedMappingJson == null) {
+                    List<String> indexMappings = NeuralSearchClusterUtil.instance().getIndexMapping(request.indices());
+                    String combinedMapping = String.join(", ", indexMappings);
+                    cachedMappingJson = gson.toJson(combinedMapping);
+                    indexMappingCache.put(cacheKey, cachedMappingJson);
+                }
+
+                parameters.put("index_mapping", cachedMappingJson);
+            } catch (Exception e) {
+                log.warn("Failed to get index mapping", e);
+            }
+        }
+
+        if (agenticQuery.getQueryFields() != null && !agenticQuery.getQueryFields().isEmpty()) {
+            parameters.put("query_fields", String.join(",", agenticQuery.getQueryFields()));
+        }
+
+        mlClient.executeAgent(agentId, parameters, ActionListener.wrap(agentResponse -> {
+            try {
+                log.info("Generated Query: [{}]", agentResponse);
+
+                // Parse the agent response to get the new search source
+                BytesReference bytes = new BytesArray(agentResponse);
```

Comment:
my understanding we're fully relying on the agent for correctness of the returned query. This can be risky, also how about the query complexity and size, should we allow execution of any query?
It's ok if this is experimental, but at least add a new issue and TODO so we can return to this later. Main areas should be: query validation, clarification if query is expensive, maybe CB or similar mechanism to prevent system crushes

Reply:
Added TODO